hw02-diy-arraylist/src/main/java/my/homeworks/hw02/App.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package my.homeworks.hw02;
+import my.homeworks.hw02.DiyArrayList.DiyArrayList;
+import my.homeworks.hw02.checker.*;
+
+import java.util.ArrayList;
+
+public class App
+{
+    private ArrayList<Checker> checkers = new ArrayList<>();
```

Comment:
Лучше объявлять как интерфейсы:
```suggestion
    private List<Checker> checkers = new ArrayList<>();
```

И в других подобных местах:
```
    protected List<String> diy;

    public Checker(List<String> diy)
    {
        this.diy = diy;
    }
```
Если вы, конечно, не осознанно хотели запретить передавать туда другие реализации List, кроме указанных.